eval.go

```diff
@@ -396,6 +406,75 @@ nextFileMatch:
 	return &res, nil
 }
 
+func chunkMatchesToLineMatches(cms []ChunkMatch, contextLines int) []LineMatch {
+	lms := make([]LineMatch, 0, len(cms))
+	for _, cm := range cms {
+		lines := bytes.Split(cm.Content, []byte("\n"))
+		currentLineStart := cm.ContentStart.ByteOffset
+		for i, line := range lines {
+			var fragments []LineFragmentMatch
```

Comment:
can we preallocate with size len(ranges)? For the common case that would be correct?